[vnet] Do not pass empty ip4.addr & ip6.addr params

akhramov opened the pull request:
For vnet jails, passing either `ip4.addr` or `ip6.addr` params whether
empty or not results in following error: `vnet jails cannot have IP
address restrictions`.

This change works the aforementioned problem around by checking whether
the corresponding vectors are empty and if they are, not passing the
parameters at all.

igalic commented on src/stopped.rs:
```diff
@@ -108,34 +108,38 @@ impl StoppedJail {
+            .filter(|ip| ip.is_ipv4())
+            .map(|ip| match ip {
+                net::IpAddr::V4(ip4) => *ip4,
+                _ => panic!("unreachable"),
```
you may wanna use [`unreachable!()`](https://doc.rust-lang.org/std/macro.unreachable.html) here

akhramov replied:
Nice catch, sure!

igalic commented on src/stopped.rs:
```diff
@@ -108,34 +108,38 @@ impl StoppedJail {
+            .filter(|ip| ip.is_ipv6())
+            .map(|ip| match ip {
+                net::IpAddr::V6(ip6) => *ip6,
+                _ => panic!("unreachable"),
```
same here, use [`unreachable!()`](https://doc.rust-lang.org/std/macro.unreachable.html)

igalic reviewed (changes requested):
🙋‍♀️

igalic commented:
Re failing tests, that's on @fabianfreyer to fix

fabianfreyer commented:
Tests pass on my machine, and I don't feel like fixing Cirrus for now.

fabianfreyer commented:
Thanks for your PR @akhramov, and thanks for the review @igalic!

akhramov committed:
[vnet] Do not pass empty ip4.addr & ip6.addr params

For vnet jails, passing either `ip4.addr` or `ip6.addr` params whether
empty or not results in following error: `vnet jails cannot have IP
address restrictions`.

This change work the aforementioned problem around by checking whether
the corresponding vectors are empty and if they are, not passing the
parameters at all.
